crates/bevy_asset/src/io/file/mod.rs

```diff
@@ -73,7 +74,15 @@ impl FileAssetWriter {
     /// watching for changes.
     ///
     /// See `get_base_path` below.
-    pub fn new<P: AsRef<Path>>(path: P) -> Self {
+    pub fn new<P: AsRef<Path> + std::fmt::Debug>(path: P, should_create: bool) -> Self {
+        if should_create {
```

Comment:
This is slightly clearer
```suggestion
    pub fn new<P: AsRef<Path> + std::fmt::Debug>(path: P, create_root: bool) -> Self {
        if create_root {
```